tests/p2p_sync_throttle_test.py

```diff
@@ -0,0 +1,120 @@
+
+    accounts=createAccountKeys(2)
+    if accounts is None:
+        Utils.errorExit("FAILURE - create keys")
+
+    accounts[0].name="tester111111"
+    accounts[1].name="tester222222"
+
+    account1PrivKey = accounts[0].activePrivateKey
+    account2PrivKey = accounts[1].activePrivateKey
+
+    testWalletName="test"
+
+    Print("Creating wallet \"%s\"." % (testWalletName))
+    testWallet=walletMgr.create(testWalletName, [cluster.eosioAccount,accounts[0],accounts[1]])
+
+    # create accounts via eosio as otherwise a bid is needed
+    for account in accounts:
+        Print("Create new account %s via %s" % (account.name, cluster.eosioAccount.name))
+        trans=nonProdNode.createInitializeAccount(account, cluster.eosioAccount, stakedDeposit=0, waitForTransBlock=True, stakeNet=1000, stakeCPU=1000, buyRAM=1000, exitOnError=True)
+        transferAmount="100000000.0000 {0}".format(CORE_SYMBOL)
+        Print("Transfer funds %s from account %s to %s" % (transferAmount, cluster.eosioAccount.name, account.name))
+        nonProdNode.transferFunds(cluster.eosioAccount, account, transferAmount, "test transfer", waitForTransBlock=True)
+        trans=nonProdNode.delegatebw(account, 20000000.0000, 20000000.0000, waitForTransBlock=True, exitOnError=True)
+
+    beginLargeBlocksHeadBlock = nonProdNode.getHeadBlockNum()
+
+    Print("Configure and launch txn generators")
+    targetTpsPerGenerator = 100
+    testTrxGenDurationSec=60
+    trxGeneratorCnt=1
+    cluster.launchTrxGenerators(contractOwnerAcctName=cluster.eosioAccount.name, acctNamesList=[accounts[0].name,accounts[1].name],
+                                acctPrivKeysList=[account1PrivKey,account2PrivKey], nodeId=prodNode.nodeId, tpsPerGenerator=targetTpsPerGenerator,
+                                numGenerators=trxGeneratorCnt, durationSec=testTrxGenDurationSec, waitToComplete=True)
+
+    endLargeBlocksHeadBlock = nonProdNode.getHeadBlockNum()
+
+    throttlingNode = cluster.unstartedNodes[0]
+    i = throttlingNode.cmd.index('--p2p-listen-endpoint')
+    throttlingNode.cmd[i+1] = throttlingNode.cmd[i+1] + ':40000B/s'
+
+    cluster.biosNode.kill(signal.SIGTERM)
+    clusterStart = time.time()
+    cluster.launchUnstarted(2)
+
+    throttledNode = cluster.getNode(3)
+    time.sleep(15)
+    assert throttlingNode.waitForBlock(endLargeBlocksHeadBlock), f'wait for block {endLargeBlocksHeadBlock}  on throttled node timed out'
+    endThrottlingSync = time.time()
+    assert throttledNode.waitForBlock(endLargeBlocksHeadBlock, timeout=90), f'Wait for block {endLargeBlocksHeadBlock} on sync node timed out'
+    endThrottledSync = time.time()
+    Print(f'Unthrottled sync time: {endThrottlingSync - clusterStart} seconds')
+    Print(f'Throttled sync time: {endThrottledSync - clusterStart} seconds')
+    assert endThrottledSync - clusterStart > endThrottlingSync - clusterStart + 15, 'Throttled sync time must be at least 15 seconds greater than unthrottled'
```

Comment:
please add comment with explanation of how did you calculate those 15 seconds

Reply:
Added comment.

Reply:
I was expecting something like (avg_block_size * block_amount) / throttle_size_per_sec.

Reply:
I don't want to give the illusion of precision where there is none.  The time to synchronize varies significantly depending on the machine running the test.  I tried to choose values which are reasonable for a wide range of machines in order to make the test as reliable as possible, but that necessarily involves some guesswork.  Unfortunately the actual block sizes are not available to the test framework so I can't implement my preferred solution and just calculate values.

Reply:
well, I understand you point but that makes this not a real test but guessed numbers that make this script pass. I understand that you can't do this precisely but if you can think of any approximate formula it would be nice. e.g. you can roughly check blocks.log size and divide this by number of blocks or you can take it from network stats that node supposed to print, to include all messages and serialized data sizes.
Otherwise, imagine this test fails. How do I know if that is error or just block size has changed or network issues? To solve this I can play with numbers to make it pass but I still have no idea if I'm not hiding some bug that way.